extensions/tmtv/src/Panels/PanelROIThresholdSegmentation.tsx

```diff
@@ -0,0 +1,456 @@
+              <Button
+                className="px-2 py-2 text-base"
+                onClick={() => {
+                  runCommand('createTMTVRTReport');
+                }}
+                disabled={tmtvValue === null}
+              >
+                {t('Create RT Report')}
+              </Button>
+            </ButtonGroup>
+          </div>
+        ) : null}
+      </div>
+    </div>
+  );
+}
+
+PanelRoiThresholdSegmentation.propTypes = {
+  commandsManager: PropTypes.shape({
+    runCommand: PropTypes.func.isRequired,
+  }),
+  servicesManager: PropTypes.shape({
+    services: PropTypes.shape({
+      SegmentationService: PropTypes.shape({
+        getSegmentation: PropTypes.func.isRequired,
+        getSegmentations: PropTypes.func.isRequired,
+        toggleSegmentationsVisibility: PropTypes.func.isRequired,
+        subscribe: PropTypes.func.isRequired,
+        EVENTS: PropTypes.object.isRequired,
+        VALUE_TYPES: PropTypes.object.isRequired,
+      }).isRequired,
+    }).isRequired,
+  }).isRequired,
+};
+
+function onSegmentationItemEditHandler({
+  id,
+  SegmentationService,
+  UIDialogService,
+}) {
+  const segmentation = SegmentationService.getSegmentation(id);
+
+  const onSubmitHandler = ({ action, value }) => {
+    switch (action.id) {
+      case 'save': {
+        SegmentationService.addOrUpdateSegmentation(
+          id,
+          {
+            ...segmentation,
+            ...value,
+          },
+          true
+        );
+      }
+    }
+    UIDialogService.dismiss({ id: 'enter-annotation' });
+  };
+
+  UIDialogService.create({
+    id: 'enter-annotation',
+    centralize: true,
+    isDraggable: false,
+    showOverlay: true,
+    content: Dialog,
+    contentProps: {
+      title: 'Enter your annotation',
+      noCloseButton: true,
+      value: { label: segmentation.label || '' },
+      body: ({ value, setValue }) => {
+        const onChangeHandler = event => {
+          event.persist();
+          setValue(value => ({ ...value, label: event.target.value }));
+        };
+
+        const onKeyPressHandler = event => {
+          if (event.key === 'Enter') {
+            onSubmitHandler({ value, action: { id: 'save' } });
+          }
+        };
+        return (
+          <div className="p-4 bg-primary-dark">
+            <Input
+              autoFocus
+              className="mt-2 bg-black border-primary-main"
+              type="text"
+              containerClassName="mr-2"
+              value={value.label}
+              onChange={onChangeHandler}
+              onKeyPress={onKeyPressHandler}
+            />
+          </div>
+        );
+      },
+      actions: [
+        // temp: swap button types until colors are updated
+        { id: 'cancel', text: 'Cancel', type: 'primary' },
+        { id: 'save', text: 'Save', type: 'secondary' },
+      ],
+      onSubmit: onSubmitHandler,
+    },
+  });
+}
+
+function ROIThresholdConfiguration({ config, setConfig, runCommand }) {
```

Comment:
I feel like it would be good to break up some of these components into multiple files, there is a lot of JSX here.